lib/core/test/unit/Cardano/Wallet/Api/ServerSpec.hs

```diff
@@ -265,3 +317,116 @@ errorHandlingSpec = describe "liftHandler and toServerError" $ do
 
         runHandler (getNetworkClock (error "bomb") True)
             `shouldReturn` Left expectedErr
+
+listStakeKeysSpec :: Spec
+listStakeKeysSpec = do
+    describe "Listing stake keys" $ do
+        it "no rewards => sum of all keys' stake == utxo balance" $ do
+            property prop_listStakeKeysBalance
+        it "ours and foreign keys are disjoint" $ do
+            property prop_listStakeKeysDisjoint
+
+prop_listStakeKeysDisjoint
+    :: UTxO
+    -> Set RewardAccount
+    -> (Map RewardAccount (Maybe Coin))
+    -> Property
+prop_listStakeKeysDisjoint utxo ours' m =
+    let
+        -- Build a list to pass listStakeKeys'
+        ours = zipWith (\ix acc -> (acc, ix, noDelegation))
+            [(0::Natural) ..]
+            (Set.toList ours')
+
+        ApiStakeKeys{_ours, _foreign, _none } = runIdentity $
+            listStakeKeys' utxo accountOfAddress (const $ pure m) ours
+
+        ourKeys = keys _ours
+        foreignKeys = keys _foreign
+    in
+        classify (length ourKeys > 1 && length foreignKeys > 1) "non-trivial" $
+            keys _ours `disjoint` keys _foreign
+  where
+    labeledCheck notF f a b = counterexample
+        (show a <> " " <> notF <> " " <> show b)
+        (property $ f a b)
+
+    disjoint = labeledCheck "is not disjoint from" Set.disjoint
+
+    keys = Set.fromList . map (view #_key)
+
+prop_listStakeKeysBalance
+    :: UTxO
```

Comment:
```suggestion
    :: Blind UTxO
```
I recall that the output of `show` for values of type `UTxO` can be quite verbose and hard-to-read, particularly if there are multiple UTxO entries.

Perhaps you could use something like [`pShow`](https://hackage.haskell.org/package/pretty-simple-4.0.0.0/docs/Text-Pretty-Simple.html#v:pShow) to render them in a clearer way on failure (perhaps as a counterexample string).

Reply:
Yeah, good idea, thanks for the exact link.

A `PrettyShow` wrapper seems better to me though -> 580b801c601765522f1cdd04adb2159c41897935